sphinx/source/conf.py

```diff
@@ -201,3 +208,65 @@
 
 # If true, `todo` and `todoList` produce output, else they produce nothing.
 todo_include_todos = True
+
+
+# -- Docstring Improvements --------------------------------------------------
+
+
+def autodoc_process_docstring(
+    app, what: str, name: str, obj, options, lines: List[str]
+) -> None:
+    """
+    Modify docstrings before creating html files.
+
+    Sphinx converts the 'Args:' and 'Returns:' sections of docstrings into
+    reStructuredText (rST) syntax, which can then be found via ':type' & ':rtype'.
+
+    See here for more information:
+    https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html
+    """
+    for i in range(len(lines)):
+        # Skip unless line is an parameter doc or a return doc
+        if not (lines[i].startswith(":type") or lines[i].startswith(":rtype")):
+            continue
```

Comment:
@aobo-y It only replaces types inside the brackets, so in your example on the `int` would be replaced. The variable description is left unchanged.

Reply:
@aobo-y 

Before Sphinx does anything, docstrings are converted into the Sphinx format for use internally. This format separates out the descriptions and types. For your example, Sphinx sees it as:

```
    :param input: blah blah ... Any values blah blah ...
    :type input: int
```

You can see other examples of the Sphinx docstring format here: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html


This conversion happens because Captum uses Sphinx's `sphinx.ext.napoleon` extension:

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/

From the Napoleon link:

> Napoleon is a pre-processor that parses [NumPy](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) and [Google](http://google.github.io/styleguide/pyguide.html#Comments) style docstrings and converts them to reStructuredText before Sphinx attempts to parse them. This happens in an intermediate step while Sphinx is processing the documentation, so it doesn’t modify any of the docstrings in your actual source code files.